recipes/openssl/3.x.x/test_v1_package/CMakeLists.txt

```
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+option(OPENSSL_WITH_ZLIB "OpenSSL with zlib support" ON)
+option(OPENSSL_WITH_LEGACY "OpenSSL with support for the legacy provider" ON)
+option(OPENSSL_WITH_MD4 "OpenSSL with MD4 support (needs legacy provider)" ON)
+option(OPENSSL_WITH_RIPEMD160 "OpenSSL with RIPEMD16 support (needs legacy provider)" ON)
+
+set(OpenSSL_DEBUG 1)
+find_package(OpenSSL REQUIRED)
+
+# Test whether variables from https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
+# are properly defined in conan generators
+set(_custom_vars
+    OPENSSL_FOUND
+    OPENSSL_INCLUDE_DIR
+    OPENSSL_CRYPTO_LIBRARY
+    OPENSSL_CRYPTO_LIBRARIES
+    OPENSSL_SSL_LIBRARY
+    OPENSSL_SSL_LIBRARIES
+    OPENSSL_LIBRARIES
+    OPENSSL_VERSION
+)
+foreach(_custom_var ${_custom_vars})
+    if(DEFINED ${_custom_var} AND NOT "${${_custom_var}}" STREQUAL "" )
+        message(STATUS "${_custom_var}: ${${_custom_var}}")
+    else()
+        message(FATAL_ERROR "${_custom_var} not defined")
+    endif()
+endforeach()
+
+add_executable(digest ../test_package/digest.c)
+if(OPENSSL_WITH_ZLIB)
+    target_compile_definitions(digest PRIVATE WITH_ZLIB)
+endif()
+target_link_libraries(digest OpenSSL::Crypto)
+
+if(OPENSSL_WITH_LEGACY)
+    add_executable(digest_legacy ../test_package/digest_legacy.c)
+    # do now show deperecation warnings
+    target_compile_definitions(digest_legacy PRIVATE OPENSSL_SUPPRESS_DEPRECATED)
+    if(OPENSSL_WITH_MD4)
+      target_compile_definitions(digest_legacy PRIVATE OPENSSL_WITH_MD4)
+    endif()
+    if(OPENSSL_WITH_RIPEMD160)
+      target_compile_definitions(digest_legacy PRIVATE OPENSSL_WITH_RIPEMD160)
+    endif()
+    if(OPENSSL_WITH_ZLIB)
+        target_compile_definitions(digest_legacy PRIVATE WITH_ZLIB)
+    endif()
+    target_link_libraries(digest_legacy OpenSSL::Crypto)
+endif()
```

```suggestion
cmake_minimum_required(VERSION 3.1)
project(test_package C)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```

Re-use the CMake + C source from `test_package` as much as possible